app/components/Notifications/NotificationContainer/index.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useCallback } from 'react';
+import {
+    _cs,
+    isDefined,
+} from '@togglecorp/fujs';
+import { gql, useMutation } from '@apollo/client';
+import {
+    IoChevronDown,
+    IoChevronUp,
+    IoCheckmark,
+    IoArrowUndoSharp,
+} from 'react-icons/io5';
+import {
+    Button,
+    QuickActionButton,
+    useAlert,
+    Container,
+    DateOutput,
+} from '@the-deep/deep-ui';
+
+import Avatar from '#components/Avatar';
+import { useModalState } from '#hooks/stateManagement';
+
+import {
+    NotificationStatusUpdateMutation,
+    NotificationStatusUpdateMutationVariables,
+} from '#generated/types';
+
+import { Notification } from '../types';
+import styles from './styles.css';
+
+const NOTIFICATION_STATUS_UPDATE = gql`
+mutation NotificationStatusUpdate($notificationId: String!, $newStatus: status!) {
+    notificationStatusUpdate(data: { id: $notificationId, status: $newStatus }) {
+        ok
+        result {
+            id
+            notificationType
+            notificationTypeDisplay
+            data
+            project {
+                id
+                title
+            }
+            status
+            timestamp
+        }
+    }
+}
+`;
+
+interface Props {
+    className?: string;
+    content?: React.ReactNode;
+    userName: string;
+    notification: Notification;
+    description?: string;
+    descriptionLabel?: string;
+}
+
+function NotificationContainer(props: Props) {
+    const {
+        content,
+        className,
+        userName,
+        notification: {
+            id: notificationId,
+            timestamp,
+            status,
+        },
+        description,
+        descriptionLabel,
+    } = props;
+
+    const [
+        isDescriptionShown,,,,
+        toggleDescriptionVisibility,
+    ] = useModalState(false);
+
+    const alert = useAlert();
+
+    const [
+        updateStatus,
+    ] = useMutation<NotificationStatusUpdateMutation, NotificationStatusUpdateMutationVariables>(
+        NOTIFICATION_STATUS_UPDATE,
+        {
+            onCompleted: (response) => {
```

Comment:
shouldn't we re-fetch notifications after update?

Reply:
This happens automatically with the help of apollo.